### DISCUSSION TITLE
Instrumenting Kubernetes operator for tracing

### DISCUSSION THREAD

**Discussion by Rajarajan97** [grafana/tempo#3655] [General]
We have seen examples of tracing for request and response scenario. But majority of work is done in backend in separate like Kubernetes custom controller(operator). How do we instrument a full operator flow.Because the operator workflow might be divided into different steps. Each time it will run one step. How to instrument it such that, we can see the whole tracing from beginning to end

@joe-elliott @electron0zero

**Comment by mapno** [comment]
If I understand correctly, and you want traces from your operator, you'll need instrumentation in the operator itself. If you're writing your own operators, you can use of the [OpenTelemetry SDKs](https://opentelemetry.io/docs/languages/) for instrumentation

**Comment by Rajarajan97** [comment]
I'm writing my own operator. We have a workflow that we have broken it into multiple steps in the operator(using switch case or if blocks).We store the current step in one of the fields in CRD. it will complete one step and reconcile it again to the operator. when it runs again it will go into the next block. How do I propagate the trace/span context in this flow? If I pick the primary trace start, it should have all the steps ran in the workflow. How can we do that using telemetry?
<img width="359" alt="Screenshot 2024-05-08 at 4 40 39 PM" src="https://github.com/grafana/tempo/assets/42295430/6e8b240b-2d9d-4334-8ae8-ee3a890812d2">

@mapno

**Comment by joe-elliott** [comment]
This seems like a question about your code structure which it's tough to give advice on. You would likely to hold a go context and pass it to the individual steps as they are executing. That's easy to say and may be harder to accomplish.
